internal/tiltfile/tiltfile_state.go

```diff
@@ -1106,13 +1107,23 @@ func (s *tiltfileState) translateK8s(resources []*k8sResource) ([]model.Manifest
 			ResourceDependencies: mds,
 		}
 
-		nonWorkload := false
-		if r.manuallyGrouped {
-			nonWorkload = len(r.extraPodSelectors) == 0 && len(r.dependencyIDs) == 0
+		podReadinessMode := r.podReadinessMode
+		if podReadinessMode == model.PodReadinessNone {
+			// Auto-infer the readiness mode
+			podReadinessMode = model.PodReadinessWait
+
+			// If the resource was
+			// 1) manually grouped (i.e., we didn't find any images in it)
+			// 2) doesn't have pod selectors, and
+			// 3) doesn't depend on images
+			// assume that it will never create pods.
+			if r.manuallyGrouped && len(r.extraPodSelectors) == 0 && len(r.dependencyIDs) == 0 {
```

Comment:
This still feels overloaded to me, but definitely better than what came before it.

Reply:
ya...let me think about this a bit more before merging immediately

Reply:
spitballing a bit here, i think whenever we define a new CRD, we need to know:

- how to find the images in it
- how to find any pods it deploys (if they can't be found by owner references)
- if it should not expect pods at all (e.g., PostgresVersion)
- if it should wait for the pods to be ready before building the next resource (e.g., servers)
- if it should wait for the pods to be complete before building the next resource (e.g., jobs)

and it's complicated a bit by the fact that there are both normal CRDs where the image shows up in the same place each time, and more meta CRDs (like HelmRelease) where it might appear in different places

i think we should at least have a way to have per-kind pod readiness configs (iin addition to a per-resource pod readiness config)

Reply:
ooh yeah per kind would be interesting to explore.